firestore/src/android/field_value_android.cc

```diff
@@ -176,63 +191,65 @@ FieldValueInternal::FieldValueInternal(const MapFieldValue& value)
     Local<String> key = env.NewStringUtf(kv.first);
     map.Put(env, key, ToJava(kv.second));
   }
-  object_ = map;
+  object_ = ArenaRef(env, map);
 }
 
 Type FieldValueInternal::type() const {
   if (cached_type_ != Type::kNull) {
     return cached_type_;
   }
-  if (!object_) {
+
+  Env env;
+  if (!object_.get(env)) {
     return Type::kNull;
   }
 
   // We do not have any knowledge on the type yet. Check the runtime type with
   // each known type.
-  Env env = GetEnv();
-  if (env.IsInstanceOf(object_, Boolean::GetClass())) {
+  if (env.IsInstanceOf(object_.get(env), Boolean::GetClass())) {
```

Comment:
Similarly, I think we should just override IsInstanceOf to take ArenaRef as well, like this:

```
bool Env::IsInstanceOf(const ArenaRef& object, const Class& clazz) {
  return IsInstanceOf(object.get(*this), clazz);
}
```
```

Reply:
Do I need to write tests for this method in `arena_ref_android_test.cc`?

Reply:
We should add a test file for env and just add tests for the new stuff that you've added to env.h.